src/Summoner/License.hs

```diff
@@ -42,6 +43,23 @@ instance Show LicenseName where
     show Apache20 = "Apache-2.0"
     show MPL20    = "MPL-2.0"
 
+instance Read LicenseName where
+    readPrec = TR.choice $ tryParse
+        [ ("MIT", MIT), ("BSD2", BSD2)
```

Comment:
You should implement `Read` instance using `parseLicense` function. The key point of `parseLicense` is to write mapping between `LicenseName` and `String` only once and then reuse it. This makes code more robust to refactoring.